src/modules/blockly/generators/propc/sensors.js

```diff
@@ -2072,17 +2077,35 @@ Blockly.Blocks.lsm9ds1_tilt = {
 };
 
 /**
- *
+ * IMU tilt block C code generator
  * @return {string}
  */
 Blockly.propc.lsm9ds1_tilt = function() {
   let code = '';
+  // TODO: Refactor getAllBlocks reference to GetBlocksByType
+  // const blocks = Blockly
+  //     .getMainWorkspace()
+  //     .getBlocksByType('IMU initialize', false);
+  // console.log(blocks);
+
   const allBlocks = Blockly.getMainWorkspace().getAllBlocks().toString();
+
   if (allBlocks.indexOf('IMU initialize') === -1) {
     code += '// ERROR: Missing IMU initialize block!';
   } else {
-    const t1Axis = '__imu' + this.getFieldValue('A1')[6].toUpperCase();
```

Comment:
The call to getFieldValue can return a null or undefined value. In that case, a TypeError is thrown when the dereference to [6] is made. The replacement code tests for this possibility before an attempt is made to use the string.